src/cpp/rtps/transport/TCPTransportInterface.cpp

```diff
@@ -242,6 +242,14 @@ void TCPTransportInterface::clean()
         }
     }
 
+    if (initial_peer_local_locator_socket_ != nullptr)
+    {
+        if (initial_peer_local_locator_socket_->is_open())
+        {
+            initial_peer_local_locator_socket_->close();
+        }
+    }
```

Comment:
We might need to enforce deletion of the socket depending on the platform

```suggestion

        initial_peer_local_locator_socket_.reset();
    }
```